### PR TITLE
[v.next] Support new grid

### PR DESCRIPTION
**Pull request by Herst**
### Changes

Ports over the following checks:

- [W009](https://github.com/twbs/bootlint/wiki/W009) (empty spacer column)
- [E005](https://github.com/twbs/bootlint/wiki/E005) (row and col on same element)
- [E012](https://github.com/twbs/bootlint/wiki/E012) (input-group and col on the same element)
- [E013](https://github.com/twbs/bootlint/wiki/E013) (invalid children of rows)
- [E014](https://github.com/twbs/bootlint/wiki/E014) (invalid parent of cols)
- [E029](https://github.com/twbs/bootlint/wiki/E029) (redundant col classes)
- [E037](https://github.com/twbs/bootlint/wiki/E037) (zero width col classes)
- [E051](https://github.com/twbs/bootlint/wiki/E051) (float style and col)
- [E052](https://github.com/twbs/bootlint/wiki/E052) (float style and row)

All of them existed already in the original Bootlint.

### Status
- **Code**: Done
- **Tests**: Done (but hopefully I didn't delete some too much, need to check)
- **Wiki**: Done

### Notes

- Simplified the empty column message generation a lot. Since the user is pointed to the Wiki it could be explained in more detail there.
- No idea what is with `.clearfix` in relationship with E013, I left it in there for now but I am not sure it is useful with the new column model (@mdo).
- I didn't adapt the wiki page examples in relationship to `.form-row`.

### CODE REVIEW

File: .idea/bootlint-ng.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Johann-S) [inline .idea/bootlint-ng.iml]: Please do not push your `.idea` folder

File: test/fixtures/grid/col-xl-0.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<!doctype html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>Test</title>
+        <!--[if lt IE 9]>
```
Comment (Johann-S) [inline test/fixtures/grid/col-xl-0.html]: not usefull on BS4

Comment (Herst) [reply test/fixtures/grid/col-xl-0.html]: True (see twbs/bootstrap#21476) but not relevant for this PR. IMHO, we should track such little details in some issue and then do it all at once at the end.

File: .idea/bootlint-ng.iml (hunk 1 above)
Comment (Herst) [reply .idea/bootlint-ng.iml]: Whoops

File: test/fixtures/grid/col-xl-0.html (hunk 2 above)
Comment (Herst) [reply test/fixtures/grid/col-xl-0.html]: #421

File: src/bootlint.js [hunk 3]
```diff
@@ -15,22 +15,23 @@ var LocationIndex = _location.LocationIndex;
...
+        'sm': 1,
+        'md': 2,
+        'lg': 3,
+        'xl': 4
     };
-    var NUM2SCREEN = ['xs', 'sm', 'md', 'lg'];
+    var NUM2SCREEN = ['', 'sm', 'md', 'lg', 'xl'];
     var IN_NODE_JS = Boolean(cheerio.load);
     var MIN_JQUERY_VERSION = '1.9.1';   // as of Bootstrap v3.3.0
     var CURRENT_BOOTSTRAP_VERSION = '3.3.7';
```
Comment (hebbet) [inline src/bootlint.js]: should be changed.

Comment (Herst) [reply src/bootlint.js]: Not part of this PR. At first I planned on dropping it (in #419) but I guess we will adapt it for Bootstrap 4+ after all, just as part of this PR.

**Pull request merged**